edb/common/traceback.py

```diff
@@ -0,0 +1,192 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2009-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# mypy: disable-error-code="attr-defined"
+
+"""
+Provides stack trace formatting that prints `{filename}:{line}`, instead of
+`"{filename}", line {line}`.
+
+Stolen from Python's traceback module.
+"""
```

Comment:
Since this code is taken from CPython, we need a relevant cpython copyright notice.

https://github.com/edgedb/edgedb/blob/master/edb/tools/profiling/tracing_singledispatch.py#L4-L5 is what we do in some existing code adapted from CPython.